openedx/core/djangoapps/schedules/config.py

```diff
@@ -19,3 +24,49 @@
     'course_update_show_unsubscribe',
     __name__
 )
+
+# This experiment waffle is supporting an A/B test we are running on sending course updates from an external service,
+# rather than through platform and ACE. See ticket AA-661 for more information.
+# Don't use this flag directly, instead use the `set_up_external_course_updates` and `query_external_updates` methods
+# below. We save this flag decision at enrollment time and don't change it even if the flag changes. So you can't just
+# directly look at flag result.
+_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID = 18
+_EXTERNAL_COURSE_UPDATES_FLAG = ExperimentWaffleFlag(WAFFLE_FLAG_NAMESPACE, 'external_updates', __name__,
+                                                     experiment_id=_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID,
+                                                     use_course_aware_bucketing=False)
+
+
+def set_up_external_course_updates(user, course_key):
+    """
+    Returns and stores whether a user should be getting the "external course updates" experience.
+
+    Basically just a wrapper around our experiment waffle flag, but only buckets users that directly enrolled (rather
+    than users enrolled by staff), for technical "waffle-flags-can-only-get-the-user-from-the-request" reasons
+
+    This saves the decision in experiment data tables. It is also idempotent and will not change after the first
+    call for a given user/course, regardless of how the waffle answer changes.
+    """
+    request = get_current_request()
+    user_is_valid = request and hasattr(request, 'user') and request.user.id and request.user.id == user.id
+    experiment_on = _EXTERNAL_COURSE_UPDATES_FLAG.is_experiment_on(course_key)
+    if user_is_valid and experiment_on:
+        # Don't send tracking info as it might differ from our saved value, and we already send the bucket in
+        # enrollment segment events.
+        bucket = _EXTERNAL_COURSE_UPDATES_FLAG.get_bucket(course_key, track=False)
+    else:
+        bucket = -1  # a special value meaning to ignore this enrollment for analytics purposes
+
+    data, _created = ExperimentData.objects.get_or_create(experiment_id=_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID,
+                                                          user_id=user.id, key=str(course_key),
+                                                          defaults={'value': str(bucket)})
+    return int(data.value)
+
+
+def query_external_updates(user_id, course_id):
```

Comment:
i'm still not sure what external_updates are but should this method be called something like

should_see_external_updates or just has_external_updates and you can directly return a Boolean for the query

Reply:
> i'm still not sure what external_updates are but should this method be called something like

The comment above the waffle didn't help? It just means "don't send nudge/update emails for them, an external service will handle it"

> should_see_external_updates or just has_external_updates and you can directly return a Boolean for the query

Can't just return a Boolean, because this is designed to be used in `receivers.py` as part of a query.

Reply:
ah you can't just do this?
```
external_updates_enabled=should_see_external_updates(OuterRef('enrollment__user_id'), OuterRef('enrollment__course_id'))
```

Reply:
You mean use that line there, but with the query cast to a Boolean inside `should_see_external_updates`?

That would resolve the query immediately, but we are trying to write a subquery plugged into a larger query (with `Exists` which does that same coercion into a bool, but lazily as part of the mysql query). Resolving the query immediately by casting to a Boolean would likely cause an exception as the OuterRefs complain that they can't find their outer ref.

Reply:
ah that makes sense, thanks! feel free to ignore all my naming comments per my comment above

Reply:
The docstring might help make this clearer now?

Reply:
yup!